### PR TITLE
test(info): add unit tests for system/environment info module

### PR DESCRIPTION
**Pull request by TimeToBuildBob**
## Summary
- Add **94 unit tests** for `gptme/info.py` — a 447-line module used by `--version` and `gptme-doctor` with **zero previous test coverage**
- Tests cover all 12 public functions and 2 internal helpers across 10 test classes

## Test Categories

| Category | Tests | Coverage |
|----------|-------|----------|
| ExtraInfo/InstallInfo dataclasses | 5 | Construction, defaults, field isolation |
| `_is_package_installed` | 10 | Name resolution, namespace packages, hyphens/underscores, error handling |
| `_parse_extras_from_metadata` | 14 | Metadata parsing, internal extras filtering, requirement formats, dedup |
| `_get_extras` caching | 1 | Cache hit behavior |
| `get_install_info` | 10 | Installer detection (uv/pip/pipx/poetry), editable, PathDistribution |
| `get_installed_extras` | 4 | Installation status, empty packages, any-match semantics |
| Provider/model/tool helpers | 6 | Return values, error fallbacks |
| `get_quick_health` | 5 | Required vs optional tools, provider/config warnings |
| System/config info | 5 | Output keys, types, project config detection |
| `format_version_info` | 26 | Human-readable + JSON, verbose mode, health indicators, provider thresholds |
| Edge cases | 4 | None requires, null installer, constants validation |

## Test plan
- [x] All 94 tests pass locally (~3s)
- [x] Ruff lint clean
- [x] Mypy passes
- [ ] CI passes

### CODE REVIEW

File: tests/test_info.py [hunk 1]
```diff
@@ -0,0 +1,1119 @@
...
+            result = get_available_providers()
+            assert result == ["openai", "anthropic"]
+
+    def test_error_returns_empty(self):
+        with (
+            patch.dict("sys.modules", {"gptme.llm": None}),
+            patch("gptme.info.get_available_providers", return_value=[]),
+        ):
+            result = get_available_providers()
+            assert result == []
```
[the remark this answers is not in the document]
Comment (TimeToBuildBob) [reply tests/test_info.py:575]: Fixed in 7af870637. Replaced the vacuous double-patch with `patch('gptme.llm.list_available_providers', side_effect=Exception('error'))` which actually exercises the except-clause in `get_available_providers`.

File: tests/test_info.py [hunk 2]
```diff
@@ -0,0 +1,1119 @@
...
+
+
+# ─── get_available_providers tests ───────────────────────────────────────────
+
+
+class TestGetAvailableProviders:
+    """Tests for get_available_providers."""
+
+    @patch("gptme.info.get_available_providers.__module__", "gptme.info")
+    def test_returns_provider_names(self):
```
[the remark this answers is not in the document]
Comment (TimeToBuildBob) [reply tests/test_info.py]: Fixed in 7af870637. Removed the no-op `@patch` decorator.

File: tests/test_info.py [hunk 3]
```diff
@@ -0,0 +1,1119 @@
...
+
+    @patch("gptme.config.get_project_config")
+    def test_project_config_detected(self, mock_project_cfg, tmp_path):
+        """When gptme.toml exists, project_config should be set."""
+        (tmp_path / "gptme.toml").write_text("[prompt]\n")
+        mock_project_cfg.return_value = MagicMock()
+        with patch("gptme.info.Path.cwd", return_value=tmp_path):
+            info = get_config_info()
+            if "project_config" in info:
+                assert "gptme.toml" in info["project_config"]
```
[the remark this answers is not in the document]
Comment (TimeToBuildBob) [reply tests/test_info.py]: Fixed in 7af870637. Changed to two hard `assert` statements so the test fails if `project_config` is missing.

File: tests/test_info.py [hunk 4]
```diff
@@ -0,0 +1,1119 @@
...
+            result = _is_package_installed("opentelemetry-api")
+            assert result is True
+            mock_find.assert_any_call("opentelemetry")
+
+    def test_underscore_name_tries_variant(self):
+        """Package with underscore should try underscore variant too."""
+        with patch("importlib.util.find_spec") as mock_find:
+            mock_find.side_effect = [None, MagicMock()]
+            result = _is_package_installed("my_package")
+            assert result is True
```
[the remark this answers is not in the document]
Comment (TimeToBuildBob) [reply tests/test_info.py]: Fixed in c470ac1fe. Corrected the source: the underscore variant is now only appended when it differs from the original name (condition changed from `'-' in name or '_' in name` to `'-' in name`). For `'my_package'`, `find_spec` is now called exactly once. Test updated accordingly.

**Comment by TimeToBuildBob** [comment]
@greptileai review

File: tests/test_info.py [hunk 5]
```diff
@@ -0,0 +1,1118 @@
...
+                ExtraInfo(name="test", installed=False, description="test")
+            ]
+            result1 = gptme.info._get_extras()
+            result2 = gptme.info._get_extras()
+            # Should only parse once
+            mock_parse.assert_called_once()
+            assert result1 is result2
+
+        # Clean up
+        gptme.info._EXTRAS_CACHE = None
```
[the remark this answers is not in the document]
Comment (TimeToBuildBob) [reply tests/test_info.py:354]: Fixed in a42ce06c. Wrapped the assertions and cache cleanup in a `try/finally` block so `_EXTRAS_CACHE` is always reset even if `assert_called_once()` or the identity check raises.

File: tests/test_info.py [hunk 6]
```diff
@@ -0,0 +1,1118 @@
...
+                name="datascience",
+                installed=False,
+                description="Data science",
+                packages=["numpy", "pandas", "matplotlib"],
+            )
+        ]
+        # numpy installed, others not
+        mock_installed.side_effect = [True]
+        result = get_installed_extras()
+        assert result[0].installed is True
```
[the remark this answers is not in the document]
Comment (TimeToBuildBob) [reply tests/test_info.py:552]: Fixed in a42ce06c. Changed `side_effect = [True]` to `side_effect = [True, False, False]` so all three packages have explicit return values — no longer relies on `any()` short-circuit behaviour.
